networkx/algorithms/community/tests/test_leiden.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+
+import networkx as nx
+from networkx.algorithms.community import leiden_communities, leiden_partitions
+
+# Leiden is not implemented by networkx, so only run tests in this file for
+# backends that implement Leiden.
+backend_has_leiden_communities = (
+    "not nx.config.backend_priority.algos "
+    "or nx.config.backend_priority.algos[0] not in leiden_communities.backends"
+)
+
+backend_has_leiden_partitions = (
+    "not nx.config.backend_priority.algos "
+    "or nx.config.backend_priority.algos[0] not in leiden_partitions.backends"
+)
+
+
+@pytest.mark.skipif(backend_has_leiden_communities)
+def test_modularity_increase():
+    G = nx.LFR_benchmark_graph(
+        250, 3, 1.5, 0.009, average_degree=5, min_community=20, seed=10
+    )
+    partition = [{u} for u in G.nodes()]
+    mod = nx.community.modularity(G, partition)
+    partition = nx.community.leiden_communities(G)
+
+    assert nx.community.modularity(G, partition) > mod
+
+
+@pytest.mark.skipif(backend_has_leiden_communities)
+def test_valid_partition():
+    G = nx.LFR_benchmark_graph(
+        250, 3, 1.5, 0.009, average_degree=5, min_community=20, seed=10
+    )
+    partition = nx.community.leiden_communities(G)
+
+    assert nx.community.is_partition(G, partition)
+
+
+@pytest.mark.skipif(backend_has_leiden_partitions)
+def test_partition_iterator():
+    G = nx.path_graph(15)
+    parts_iter = nx.community.leiden_partitions(G, seed=42)
+    first_part = next(parts_iter)
+    first_copy = [s.copy() for s in first_part]
+
+    # gh-5901 reports sets changing after next partition is yielded
```

Comment:
This comment must be referring to an issue brought up by louvain. Change to clarify?
```suggestion
    # check 1st part stays fixed even after 2nd iteration (like gh-5901 in louvain)
```

Reply:
Good catch! That's better :+1: